src/lib/mina.ts

```diff
@@ -773,3 +805,143 @@ function defaultNetworkState(): NetworkValue {
     nextEpochData: cloneCircuitValue(epochData),
   };
 }
+
+async function verifyAccountUpdate(
+  account: LedgerAccount,
+  accountUpdate: AccountUpdate,
+  transactionCommitments: {
+    commitment: Field;
+    fullCommitment: Field;
+  },
+  proofsEnabled: boolean
+): Promise<void> {
+  let perm = account.permissions;
+
+  let { commitment, fullCommitment } = transactionCommitments;
+
+  // we are essentially only checking if the update is empty or an actual update
+  function includesChange(val: any): boolean {
+    if (val?.constructor === Array) {
+      return !(val as Array<any>).every((v) => v == null);
+    } else {
+      return val != null;
+    }
+  }
+
+  function permissionForUpdate(key: string): string {
```

Comment:
can we type this more precisely than `string` so we can catch if we ever add any new keys here? @mitschabaude did something fancy with the account-update type to do this in a different part of snarkyjs

If that's too complex or intractable, can we at least `throw` in the default branch so we can notice (at least when we run it once) that we need to revisit this function?